### DISCUSSION TITLE
Porgramtically get supported kafka versions

### DISCUSSION THREAD

**Discussion by kingnarmer** [strimzi/strimzi-kafka-operator#7308] [Q&A]
I am looking for an automated way to get list of kafka versions supported on strimzi operator release. 

I am test instance and would love to automate the update. 

Right now I can get the operator version during cicd script but can't get kafka versions supported from command line. 

Thanks

**Comment by Frawless** [comment]
Hello, In our systemtests we use this class https://github.com/strimzi/strimzi-kafka-operator/blob/main/systemtest/src/main/java/io/strimzi/systemtest/utils/TestKafkaVersion.java during upgrade testing. It will basically collect data from `kafka-versions.yaml` from the current git repo (tests are versioned in the same way as operators). You can also get env variable `STRIMZI_KAFKA_IMAGES` from deployment yaml https://github.com/strimzi/strimzi-kafka-operator/blob/main/install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml#L56 and parse values from there.

**Comment by scholzj** [comment]
You know that you do not need to specify the version fields and the operator will automatically use the latest versions? Maybe that is an alternative for you to simply not care about the version.

**Reply by kingnarmer** [reply]
This is existing test instance , the goal is to automate upgrade and testing . Tried to upgrade to operator 0.31.0 , however it didn't upgrade kafka version.

**Reply by scholzj** [reply]
Then you have the version specified in the Kafka CR or have some other issue.

**Reply by kingnarmer** [reply]
Yes, the initial kafka cluster creation didn't take "kafka --> version --> latest" . I had to define version on spec.kafka.version.

**Reply by scholzj** [reply]
If you don't define the `.spec.kafka.version`, it will always give you the latest supported Kafka version.
